### ISSUE TITLE
Add Email checking widget to DevServices/DevUI

### ISSUE THREAD

**Issue by GregJohnStewart** [quarkusio/quarkus#28700]
### Description

It would be terribly useful to have a widget on DevServices/ dev ui that one could use to see emails that have been sent by the service.

I was thinking something inspired by the `MockMailbox` utility used for testing; just simple read-only (from dev ui) list of addresses sent to, and the emails that were sent.

This would be way easier than making devs setup their dev instance with an actual email server.

### Implementation ideas

Implementation similar to other dev service widgets.

Maybe a flag, similar to `quarkus.mailer.mock` that would help control this? Or perhaps if that is on, enable this widget?

**Comment by geoand** [comment]
cc @cescoffier

**Comment by cescoffier** [comment]
Good idea!
I wonder if I should wait for @phillip-kruger revamped dev ui first.

**Comment by GregJohnStewart** [comment]
Circling back, what's the state of things? Is the stage set? 

Another ticket was added asking for similar capabilities: https://github.com/quarkusio/quarkus/issues/32894

**Comment by geoand** [comment]
The new DevUI is in, so it's a matter of priorities now

**Comment by phillip-kruger** [comment]
@GregJohnStewart Here are the documentation for the new Dev UI if you are keen to do a PR: https://quarkus.io/version/main/guides/dev-ui-v2

**Comment by GregJohnStewart** [comment]
I can take a stab at it sure. First time for everything :)

**Comment by phillip-kruger** [comment]
@GregJohnStewart ping me on Zulip if you have questions.

**Comment by melloware** [comment]
@GregJohnStewart check out my Extension which does exactly that as a DevService with a UI: https://github.com/quarkiverse/quarkus-mailpit

@cescoffier I think this might be closeable as I am pretty sure this is what was intended.

**Comment by cescoffier** [comment]
@melloware did you integrate the mailpit UI (or at least a mail UI) in the dev UI? We want to unified experience.

**Comment by codespearhead** [comment]
@cescoffier yes, it's baked into the Dev UI as per [this comment](https://github.com/quarkusio/quarkus/issues/32894#issuecomment-1761814492).

I wonder if this should be marked as solved now that [this PR](https://github.com/quarkiverse/quarkiverse-devops/pull/185) was merged, because the dependency is from `io.quarkiverse.mailpit` instead of `io.quarkus`.

**Comment by melloware** [comment]
I think this ticket is safe to close?  Mailpit extension is very popular now to solve this issue.

**Comment by geoand** [comment]
Definitely, thanks!
